AdtSampleApp/SampleFunctionsApp/ProcessHubToDTEvents.cs

```diff
@@ -17,23 +17,27 @@ namespace SampleFunctionsApp
     // and sets the "Temperature" property of the device with the value of the telemetry.
     public class ProcessHubToDTEvents
     {
-        private static readonly string adtInstanceUrl = Environment.GetEnvironmentVariable("ADT_SERVICE_URL");
```

Comment:
Can you help me understand the rationale for not using an application setting here? We generally advise customers store configuration values in app settings for Function apps, and they enable a nice portal UI and the ability to automatically swap between staging, production, and local values

Reply:
This is all still supported through the use of the AdtConfiguration and ConfigurationBuilder in general. The configuration builder will pull in Environment variables when present, but now also allows you to use a settings file for the same setting in a dev environment, which is much more intuitive than allowing the setting only through an environment variable. Additional background can be found here: https://blog.jongallant.com/2018/01/azure-function-config/

Reply:
Aside from that, I wanted to draw attention away from managing settings and thus moved the whole logic to a "helper". That would help us highlight the steps that are really interesting in the context of accessing the ADT instance from a sample function.

Reply:
You can use application settings in that local settings file for a dev environment as well. Anything you put in the "values" field will get loaded as an app setting at startup: https://docs.microsoft.com/en-us/azure/azure-functions/functions-run-local#local-settings-file

Reply:
I'm missing how that would work if you call [Environment.GetEnvironmentVariable()](https://docs.microsoft.com/en-us/dotnet/api/system.environment.getenvironmentvariable?view=netcore-3.1). Are you saying local.settings.json overrides the process environment variables at runtime?

Reply:
Yep. It's got a whole system to load them in at runtime in the context of that process. It's pretty nifty :)

Reply:
Thanks for pointing it out. Verified and applied the change. Makes this super simple.